src/replica/split/replica_split_manager.cpp

```diff
@@ -1257,5 +1273,81 @@ void replica_split_manager::on_copy_mutation_reply(error_code ec,
     // TBD
 }
 
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica_split_manager::parent_stop_split(
+    split_status::type meta_split_status) // on parent partition
+{
+    dassert_replica(status() == partition_status::PS_PRIMARY ||
+                        status() == partition_status::PS_SECONDARY,
+                    "wrong partition_status({})",
+                    enum_to_string(status()));
+    dassert_replica(_split_status == split_status::SPLITTING ||
+                        _split_status == split_status::NOT_SPLIT,
```

Comment:
Why allow stop under `NOT_SPLIT`?

Reply:
And, if we send `cancel` or `pause` by mistake, dassert will cause crash, is your expect?

Reply:
When learn happened, parent partition may stop split, set it _split_status as `NOT_SPLIT`, meta server won't know it, so it is possible when a pause or cancel split request sync to parent partition, its split_status is NOT_SPLIT.

Reply:
> And, if we send cancel or pause by mistake, dassert will cause crash? is your expect?
Pause or cancel split request will send to meta server, meta server should check if this table is splitting, you can reference pr[679](https://github.com/XiaoMi/rdsn/pull/681). Besides, parent partition will not set split_status as PAUSING and CANCELING, when it receives pause or cancel request, it will set it NOT_SPLIT. I don't know if I explain it clearly, you can comment to me if you have any questions.